File: Data Structures & Algorithms/daily-temperatures/submission-5.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n=len(temperatures)
        res=[0]*n
        stack=[]
        for i in range(n):
            while stack and stack[-1][0]<temperatures[i]:
                res[stack[-1][1]]=i-stack[-1][1]
                stack.pop(-1)
            stack.append((temperatures[i],i))
        return res


        

File: Data Structures & Algorithms/validate-parentheses/submission-5.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        int flag=0;
        for(int i=0;i<s.size();i++)
        {
            if(s[i]=='{'||s[i]=='['||s[i]=='(')
            {
                flag=1;
                st.push(s[i]);
            }
            else if(s[i]=='}')
            {
                if(!st.empty() && st.top()=='{')
                  st.pop();
                else
                  return false;
            }
            else if(s[i]==']')
            {
                 if(!st.empty() && st.top()=='[')
                  st.pop();
                else
                  return false;
            }
            else if(s[i]==')')
            {
                 if(!st.empty() && st.top()=='(')
                  st.pop();
                else
                  return false;
            }
        }

    if(st.empty() && flag==1) return true;
    
    return false;}
};
